app/controllers/application_controller.rb

```diff
@@ -1,10 +1,14 @@
 class ApplicationController < ActionController::API
-  before_action :configure_permitted_parameters, if: :devise_controller?
+  include JwtApi
+  before_action :authenticate_request, except: %i[login register]
 
   private
 
-  def configure_permitted_parameters
-    devise_parameter_sanitizer.permit(:sign_up,
-                                      keys: %i[name email role])
+  def authenticate_request
+    header = request.headers['Authorization']
+    header = header.split.last if header
+
+    decoded = jwt_decode(header)
+    @current_user = User.find(decoded[:user_id])
   end
```

Comment:
@khanh-devos very well done by adding the jwt authentication feature 👍🏼 , I want just to ask why is this logic used here in the applicationController what if we want to open some endpoints to unauthenticated users this will bloc the access without token isn't it?

Reply:
Hello @souad988 ,
If you want to stop the authentication checking, just comment(delete) this line:
### before_action :authenticate_request, except: %i[login register]